cache/contenthash/tarsum.go

```diff
@@ -37,10 +37,10 @@ func v0TarHeaderSelect(h *tar.Header) (orderedHeaders [][2]string) {
 
 func v1TarHeaderSelect(h *tar.Header) (orderedHeaders [][2]string) {
 	pax := h.PAXRecords
-	if len(h.Xattrs) > 0 { //nolint deprecated
+	if len(h.Xattrs) > 0 { //nolint:staticcheck
```

Comment:
While we're updating; if you know _why_ we're ignoring these, it's good to add a comment describing that, e.g.

```go
//nolint:staticcheck // Ignore the flux capacitor, doc
```

Sometimes linters may flag something for multiple reasons, and it's good to know _what_ thing we're okay with ignoring.

Reply:
Aha, I really do not know for this one :eyes:

It looks quite old: https://github.com/jedevc/buildkit/blob/remove-structcheck/cache/contenthash/tarsum.go#L18-L19

I changed the `nolint` rule since `deprecated` isn't a recognized linter on https://golangci-lint.run/usage/linters/, so the deprecation warning isn't properly suppressed.

Reply:
Hm. So perhaps it's not longer needed. That said, it may be that the `Xattrs` option is deprecated, and that we currently don't have a linter warning about that. ISTR `revive` may be linting for deprecated things (but it could've been `govet`)

Reply:
It is for backwards compatibility. Even though these fields are deprecated, if we don't use them then if someone passes them to our function, we would just skip them and the caller would get unexpected behavior. I think until Go removes these fields(never for v1 iiuc) we need to support them.

Reply:
The extra comment can just say "field deprecated in stdlib"

Reply:
> The extra comment can just say "field deprecated in stdlib"

Done